### PR TITLE
fix: avoid dump drivers

### PR DESCRIPTION
**Pull request by davidbp**
Removes `_drivers ` from the `__getstate__` method. This allows executors to be loaded with  updated `_drivers` information from `.yml` files. Before this commit the information would be serialised.

### CODE REVIEW

File: tests/unit/executors/test_dump_executor_with_drivers.py [hunk 1]
```diff
@@ -23,17 +23,32 @@ def test_dump_driver(tmpdir):
     # other stateful values should be reset to init()'s time
     assert not p.idle_dealer_ids
 
+def test_dump_excutor_without_drivers(tmpdir):
```
Comment (JoanFM) [inline tests/unit/executors/test_dump_executor_with_drivers.py:26]: typo in test nane

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/executors/__init__.py [hunk 2]
```diff
@@ -149,17 +149,17 @@ def __init__(awesomeness = 5):
...
-        'query',
-        'craft',
-        'segment',
-        'score',
-        'evaluate',
-        'predict',
-        'query_by_key',
-        'delete',
-        'update',
+        "encode",
```
Comment (JoanFM) [inline jina/executors/__init__.py:152]: is this black doing it?

File: jina/executors/__init__.py [hunk 3]
```diff
@@ -27,37 +27,59 @@
     from ..peapods.runtimes.zmq.zed import ZEDRuntime
     from ..drivers import BaseDriver
 
-__all__ = ['BaseExecutor', 'AnyExecutor', 'ExecutorType']
+__all__ = ["BaseExecutor", "AnyExecutor", "ExecutorType"]
```
Comment (JoanFM) [inline jina/executors/__init__.py:30]: we try to enforce single quote style

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/executors/__init__.py (hunk 2 above)
Comment (davidbp) [reply jina/executors/__init__.py:152]: yes

File: jina/executors/__init__.py (hunk 3 above)
Comment (davidbp) [reply jina/executors/__init__.py]: done

File: jina/executors/__init__.py [hunk 4]
```diff
@@ -305,16 +334,18 @@ def save_abspath(self) -> str:
         """Get the file path of the binary serialized object
 
         The file name ends with `.bin`.
+
+        :return: the name of the file with `.bin`
         """
-        return self.get_file_from_workspace(f'{self.name}.bin')
+        return self.get_file_from_workspace(f"{self.name}.bin")
```
Comment (JoanFM) [inline jina/executors/__init__.py:340]: single quote

File: jina/executors/__init__.py [hunk 5]
```diff
@@ -305,16 +334,18 @@ def save_abspath(self) -> str:
...
 
     @property
     def config_abspath(self) -> str:
         """Get the file path of the YAML config
 
-        The file name ends with `.yml`.
+        :return: The file name ends with `.yml`.
         """
-        return self.get_file_from_workspace(f'{self.name}.yml')
+        return self.get_file_from_workspace(f"{self.name}.yml")
```
Comment (JoanFM) [inline jina/executors/__init__.py:348]: same here

File: jina/executors/__init__.py [hunk 6]
```diff
@@ -326,18 +357,22 @@ def get_shard_workspace(
...
         :param workspace_name: Name of the workspace.
         :param pea_id: Id of the pea,
-        :return: Return the workspace of the shard of this Executor.
+
+        :return: returns the workspace of the shard of this Executor.
         """
         # TODO (Joan, Florian). We would prefer not to keep `pea_id` condition, but afraid many tests rely on this
         return (
-            os.path.join(workspace_folder, f'{workspace_name}-{pea_id}')
+            os.path.join(workspace_folder, f"{workspace_name}-{pea_id}")
```
Comment (JoanFM) [inline jina/executors/__init__.py:365]: and here

File: jina/executors/__init__.py [hunk 7]
```diff
@@ -353,7 +388,7 @@ def _workspace(self):
 
         :return: returns the workspace property of the executor or default to './'
         """
-        return self.workspace or './'
+        return self.workspace or "./"
```
Comment (JoanFM) [inline jina/executors/__init__.py:391]: and here

File: jina/executors/__init__.py [hunk 8]
```diff
@@ -370,16 +405,19 @@ def get_file_from_workspace(self, name: str) -> str:
...
     @property
     def physical_size(self) -> int:
-        """Return the size of the current workspace in bytes"""
+        """Return the size of the current workspace in bytes
+
+        :return: byte size of the current workspace
+        """
         root_directory = Path(self.shard_workspace)
-        return sum(f.stat().st_size for f in root_directory.glob('**/*') if f.is_file())
+        return sum(f.stat().st_size for f in root_directory.glob("**/*") if f.is_file())
```
Comment (JoanFM) [inline jina/executors/__init__.py:420]: and here

File: jina/executors/__init__.py [hunk 9]
```diff
@@ -444,7 +486,7 @@ def save(self, filename: str = None):
             if self.max_snapshot > 0 and os.path.exists(f):
                 bak_f = (
                     f
-                    + f'.snapshot-{self._last_snapshot_ts.strftime("%Y%m%d%H%M%S") or "NA"}'
+                    + f".snapshot-{self._last_snapshot_ts.strftime('%Y%m%d%H%M%S') or 'NA'}"
```
Comment (JoanFM) [inline jina/executors/__init__.py:489]: and here

File: jina/executors/__init__.py [hunk 10]
```diff
@@ -480,13 +522,16 @@ def inject_config(
...
         :param pea_id: the id of the storage of this parallel pea
         :param read_only: if the executor should be readonly
+        :param args: Additional arguments.
+        :param kwargs: Additional key word arguments.
+
         :return: an executor object
         """
-        if 'metas' not in raw_config:
-            raw_config['metas'] = {}
+        if "metas" not in raw_config:
```
Comment (JoanFM) [inline jina/executors/__init__.py:530]: and here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/executors/__init__.py (hunk 4 above)
Comment (davidbp) [reply jina/executors/__init__.py]: updated

**Review by JoanFM — APPROVED** [review]

**Review by florian-hoenicke — APPROVED** [review]

**Pull request merged**